compiler/rustc_index/src/vec.rs

```diff
@@ -184,7 +184,7 @@ macro_rules! newtype_index {
             }
         }
 
-        unsafe impl ::std::iter::Step for $type {
+        impl ::std::iter::Step for $type {
```

Comment:
We probably want an unsafe impl TrustedStep here to avoid regressing performance?

Reply:
I'll look into this more in a bit, but at first glance it doesn't look like it's currently implemented?

Reply:
I'm not sure what you mean - we added the trusted vs. not trusted distinction in this PR and so not implementing TrustedStep on any type which used to implement Step would be a regression, correct?

Reply:
For some reason I read that as `TrustedLen`, but even in that case there was a blanket impl. You're absolutely right. I'll add a line to implement that.